### DISCUSSION TITLE
Use Querier Front-End to talk to your own prometheus query API

### DISCUSSION THREAD

**Discussion by srijan55** [cortexproject/cortex#4282] [Q&A] [answered]
I have setup my own prometheus query endpoint (with custom data source). Is it possible to setup only cortex front-end which talks to this endpoint instead of its own querier and gives me all the goodness of query caching etc.

**Comment by pstibrany** [comment, marked as answer]
Hello. Yes, that is possible. Please take a look here: https://cortexmetrics.io/docs/configuration/prometheus-frontend/

The trick is to set `downstream_url` field in the `frontend` section and only start `query-frontend` component from Cortex (using `target` field).
